### DISCUSSION TITLE
How to trigger snippets in visual mode?

### DISCUSSION THREAD

**Discussion by RHOOPH** [LazyVim/LazyVim#2249] [Q&A]
Is triggering snippets with a given selection supported by default in Lazyvim? 

For example , [This snippet](https://github.com/rafamadriz/friendly-snippets/blob/53d3df271d031c405255e99410628c26a8f0d2b0/snippets/javascript/react-ts.json#L280) is used to wrap an element with  another element. How do i trigger this?

**Comment by dpetka2001** [comment]
You should ask this in [nvim-cmp](https://github.com/hrsh7th/nvim-cmp) repo, the plugin that offers auto-completion together with Luasnip as its snippet engine (other engines can be used as well). 

As far as I can tell this is currently not possible, because only `insert` and `cmdline` modes are available according to [here](https://github.com/hrsh7th/nvim-cmp/blob/main/lua/cmp/utils/api.lua#L35-L38). It doesn't seem to support `select-mode`. 

Of course I might be wrong, since I don't understand in depth how the plugin works, that's why I suggested you ask your question
over at their repo, so they can give you a more appropriate answer.
